### PR TITLE
QR code input

### PR DESCRIPTION
**Pull request by iiLubos**
image

        MMQrCodeEditor {
          title: "MMQrCodeEditor"
          placeholderText: "QR code"
          warningMsg: text.length > 0 ? "" : "Click to icon and scan the code"
          onEditorValueChanged: function(newValue, isNull) { console.log("QR code: " + newValue) }
        }

### CODE REVIEW

**Comment by iiLubos** [comment]
Trying to add a rounded rectangle hole, but it seems OpacityMask is not supported yet

```
  import Qt5Compat.GraphicalEffects
      Rectangle {
        id: bg
        anchors.fill: parent
        visible: false
      }

      Rectangle {
        id: mask
        anchors.centerIn: parent
        width: 200
        height: 200
        visible: false
      }

      OpacityMask {
        anchors.fill: bg
        source: bg
        maskSource: mask
        opacity: 0.5
      }
```

File: app/qml/components/MMCodeScanner.qml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick
+import QtQuick.Controls
+import QtMultimedia
+// uncomment when using in MM
```
Comment (PeterPetrik) [inline app/qml/components/MMCodeScanner.qml:13]: ?

Comment (iiLubos) [reply app/qml/components/MMCodeScanner.qml]: Resolved. Added fake classes

File: app/qml/components/MMCodeScanner.qml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick
+import QtQuick.Controls
+import QtMultimedia
+import lc 1.0 as InputClass
+
+Drawer {
```
Comment (tomasMizera) [inline app/qml/components/MMCodeScanner.qml:15]: Is there any reason to make this a drawer? It looks more like a full-screen page to me

File: gallery/CMakeLists.txt [hunk 3]
```diff
@@ -17,12 +17,17 @@ find_package(
...
              Concurrent
              Positioning
+             Sensors
              QuickControls2
+             Network
              Svg
+             Sql
              Core
              Core5Compat
+             Multimedia
```
Comment (tomasMizera) [inline gallery/CMakeLists.txt:30]: Are all of these added lines really needed?

File: gallery/qrcodedecoder.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+{
+    Q_OBJECT
+
+    Q_PROPERTY( QObject *videoSink READ videoSink WRITE setVideoSink NOTIFY videoSinkChanged );
+
+  public:
+
+    explicit QrCodeDecoder( QObject *parent = nullptr ) {};
+
+    QObject *videoSink() {};
```
Comment (tomasMizera) [inline gallery/qrcodedecoder.h:17]: Shouldn't this method return `nullptr`?

Comment (tomasMizera) [reply gallery/qrcodedecoder.h:17]: or some `QObject *`?

File: app/qml/components/MMCodeScanner.qml (hunk 2 above)
Comment (iiLubos) [reply app/qml/components/MMCodeScanner.qml:15]: The drawer is the best item for it, I think. The Drawer for fullscreen is an item of a small-sized editor. So we don't need to push QR scanner item to any StackView

**Review by tomasMizera — APPROVED** [review]
🚀

**Pull request merged**